sound/soc/sof/intel/hda-pcm.c

```diff
@@ -118,8 +118,11 @@ int hda_dsp_pcm_hw_params(struct snd_sof_dev *sdev,
 	/* disable SPIB, to enable buffer wrap for stream */
 	hda_dsp_stream_spib_config(sdev, stream, HDA_DSP_SPIB_DISABLE, 0);
 
-	/* set host_period_bytes to 0 if no IPC position */
-	if (hda && hda->no_ipc_position)
+	/* set host_period_bytes to 0 if no wakeup mode
+	 * for SKL- platform, pcm_pointer is null, wakeup mode is only way to
+	 * update position.
+	 */
```

Comment:
This is not true, we need update position for non-wakeup mode also on BYT.

Reply:
i will update this comment like
/* set host_period_bytes to 0 if no wakeup mode
* if pcm_pointer is null, wakeup mode is expected.
*/